cmd/juju/action/run.go

```diff
@@ -189,43 +197,56 @@ func (c *runCommand) Run(ctx *cmd.Context) error {
 	}
 	defer c.api.Close()
 
-	results, err := c.enqueueActions(ctx)
+	// juju run action is behind a feature flag so we are
+	// free to not support running against an older controller
+	if c.api.BestAPIVersion() < 6 {
+		return errors.Errorf("juju run action not supported on this version of Juju")
+	}
+
+	operationId, results, err := c.enqueueActions(ctx)
 	if err != nil {
 		return errors.Trace(err)
 	}
+	numTasks := len(results)
+	if !c.background {
+		var plural string
+		if numTasks > 1 {
+			plural = "s"
+		}
+		ctx.Infof("Running operation %s with %d task%s", operationId, numTasks, plural)
+	}
 
 	var actionTag names.ActionTag
-	info := make(map[string]interface{}, len(results.Results))
-	for _, result := range results.Results {
-		if result.Error != nil {
-			return result.Error
+	info := make(map[string]interface{}, numTasks)
+	for i, result := range results {
+		if result.err != nil {
+			return result.err
 		}
-		if result.Action == nil {
-			return errors.Errorf("operation failed to enqueue on %q", result.Action.Receiver)
+		if result.task == "" {
+			return errors.Errorf("operation failed to enqueue on %q", result.receiver)
 		}
-		if actionTag, err = names.ParseActionTag(result.Action.Tag); err != nil {
+		if actionTag, err = names.ParseActionTag(result.task); err != nil {
 			return err
 		}
 
 		if !c.background {
-			ctx.Infof("Running Operation %s", actionTag.Id())
+			ctx.Infof("  - task %s on %s", actionTag.Id(), c.unitReceivers[i])
 		}
-		unitTag, err := names.ParseUnitTag(result.Action.Receiver)
-		if err != nil {
-			return err
-		}
-		info[unitTag.Id()] = map[string]string{
+		info[result.receiver] = map[string]string{
 			"id": actionTag.Id(),
 		}
 	}
+	ctx.Infof("")
 	if c.background {
-		if len(results.Results) == 1 {
-			ctx.Infof("Scheduled Operation %s", actionTag.Id())
-			ctx.Infof("Check status with 'juju show-operation %s'", actionTag.Id())
+		if numTasks == 1 {
```

Comment:
This Run method has become very long - maybe split it up a bit?

Reply:
ok